File: L1002-FindCommonCharacters.java
// https://leetcode.com/problems/find-common-characters/
class Solution {
    public List<String> commonChars(String[] words) {
        int res[] = new int[26], a[] = new int[26];
        int l = words.length;
        List<String> result = new ArrayList<>();
        int l1 = words[0].length();
        for (int i = 0; i < l1; i++) {
            res[words[0].charAt(i) - 'a'] += 1;
        }
        for (int i = 1;i < l; i++) {
            int length = words[i].length();
            for (int j = 0; j < length; j++) {
                a[words[i].charAt(j) - 'a'] += 1;
            }
            for (int j = 0; j < 26; j++) {
                res[j] = Math.min(res[j], a[j]);
                a[j] = 0;
            }
        }
        for (int i = 0; i < 26; i++) {
            while (res[i] > 0) {
                result.add((char) ('a' + i) + "");
                res[i] -= 1;
            }
        }
        return result;
    }
}

File: L1002-find_common_characters.py
# https://leetcode.com/problems/find-common-characters/
class Solution:
    def commonChars(self, words: List[str]) -> List[str]:
        a = [0 for _ in range(26)]
        res = [0 for _ in range(26)]
        ascii_a = ord('a')
        result = []
        alphabets = string.ascii_lowercase
        length = len(words)
        for letter in words[0]:
            res[ord(letter) - ascii_a] += 1
        for i in range(length):
            for letter in words[i]:
                a[ord(letter) - ascii_a] += 1
            for j in range(26):
                res[j] = min(res[j], a[j])
                a[j] = 0
        for i in range(26):
            while (res[i] > 0):
                result.append(alphabets[i])
                res[i] -= 1
        return result
